lib/draper/decorator.rb

```diff
@@ -239,22 +239,23 @@ def self.alias_object_to_object_class_name
     end
 
     def self.object_class_name
-      raise NameError if name.nil? || name.demodulize !~ /.+Decorator$/
+      return nil if name.nil? || name.demodulize !~ /.+Decorator$/
       name.chomp("Decorator")
     end
 
     def self.inferred_object_class
       name = object_class_name
-      name.constantize
-    rescue NameError => error
-      raise if name && !error.missing_name?(name)
-      raise Draper::UninferrableObjectError.new(self)
+      name_constant = name && name.safe_constantize
+      raise Draper::UninferrableObjectError.new(self) if name_constant.nil?
+
+      name_constant
```

Comment:
In `Decoratable#decorator_class` above, you are returning the constant unless it's nil and raise at the end. Here you are doing the exact opposite. For consistency purposes, I think it would make sense to follow the logic from above here as well:

```ruby
name_constant = name&.safe_constantize
return name_constant unless name_constant.nil?
raise Draper::UninferrableObjectError.new(self)
```